### PR TITLE
Disable the ability to instantiate into closed prefabs, and to move the root prefab in focus mode.

### LINKED ISSUE #6307: Prefab Focus Mode: Toggle pivot/Instantiate Prefab/Instantiate Procedural Prefab context menu options are available on prefab containers
**Describe the bug**
Toggle pivot/Instantiate Prefab/Instantiate Procedural Prefab context menu options are available on .procprefab containers. This allows editing of read-only entities.

**Pre-requisites**
1. Procedural Prefabs are enabled in AutomatedTesting\user\Registry\editorpreferences.setreg:

```
{
    "Amazon": {
        "Preferences": {
            "EnablePrefabSystem": true,
            "EnablePrefabSystemWipFeatures": true
        }
    },
    "O3DE": {
        "Preferences": {
            "Prefabs": {
                "SaveAllPrefabs": "Ask every time",
                "UseProceduralPrefabs": true
            }
        }
    }
}
```

**Steps to reproduce**
1. Launch the Editor and open a new or existing level.
2. Instantiate any .procprefab into the Entity Outliner.
3. Right-click the .procprefab container and Instantiate Prefab...

**Expected behavior**
Toggle pivot/Instantiate Prefab/Instantiate Procedural Prefab context menu options are not available on read-only entities.

**Actual behavior**
Toggle pivot/Instantiate Prefab/Instantiate Procedural Prefab context menu options are available on .procprefab containers when not in Focus Mode. If the .procprefab is in Focus Mode, the operations are unavailable.

**Found in Branch**
development@a8a96d3e66853fdd

### PR DESCRIPTION
**Pull request by AMZN-daimini**
Disables "Instantiate" actions for both regular and procedural prefabs when they would result in entities being added under a closed prefab instance.
image

Removes the "Move Up" and "Move Down" menu items on open prefab containers.
image

Also changes the labeling to the "edit" action for procedural prefabs to make it more accurate.
image

Fixes #6307

### CODE REVIEW

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp [hunk 1]
```diff
@@ -362,14 +379,34 @@ namespace AzToolsFramework
...
-                            editAction->setShortcut(QKeySequence(Qt::Key_Plus));
-                            editAction->setToolTip(QObject::tr("Edit the prefab in focus mode."));
-
-                            QObject::connect(editAction, &QAction::triggered, editAction, [selectedEntity] {
-                                ContextMenu_EditPrefab(selectedEntity);
-                            });
+                            if (s_prefabPublicInterface->IsOwnedByProceduralPrefabInstance(selectedEntity))
+                            {
+                                // Inspect Prefab
+                                QAction* editAction = menu->addAction(QObject::tr("Inspect Procedural Prefab"));
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:385]: I'm ok with this for now, but keep an eye out here, we might want to DRY the action creation / shortcut assignment and just update the labels.

**Review by NicholasVanSickle — APPROVED** [review]
Looks good to me, couple nits.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp (hunk 1 above)
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp:385]: Agreed. In general, I'd want to simplify the whole process and have a better system to determine which menu items appear on a specific entity. Current code is very messy and editing it is quite error-prone.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp [hunk 2]
```diff
@@ -609,23 +616,27 @@ namespace AzToolsFramework
             {
                 AZ::EntityId entityId = m_selectedEntityIds[0];
 
-                AZ::EntityId parentId;
-                EditorEntityInfoRequestBus::EventResult(parentId, entityId, &EditorEntityInfoRequestBus::Events::GetParent);
+                // Don't allow to move the entity if it's the focus root.
```
[the remark this answers is not in the document]
Comment (AMZN-daimini) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerWidget.cpp:619]: Fixed!

**Review by amzn-mike — APPROVED** [review]

**Pull request merged**